rust/lance/src/index/prefilter.rs

```diff
@@ -149,12 +179,40 @@ impl PreFilter {
     /// This method must be called after `wait_for_ready`
     #[instrument(level = "debug", skip_all)]
     pub fn filter_row_ids(&self, row_ids: &[u64]) -> Vec<u64> {
-        let block_list = self.block_list.get_ready();
-        row_ids
-            .iter()
-            .enumerate()
-            .filter(|(_, row_id)| !block_list.contains(**row_id))
-            .map(|(idx, _)| idx as u64)
-            .collect::<Vec<_>>()
+        let enumerated_ids = row_ids.iter().enumerate();
+        match (&self.block_list, &self.allow_list) {
+            (Some(block_list), Some(allow_list)) => {
+                // Only take rows that are both in the allow list and not in the block list
+                let block_list = block_list.get_ready();
+                let allow_list = allow_list.get_ready();
+                enumerated_ids
+                    .filter(|(_, row_id)| {
+                        !block_list.contains(**row_id) && allow_list.contains(**row_id)
```

Comment:
I wonder: would the `allow_list` filter be typically more selective? And if so, should we do that first?

Reply:
I went ahead and put it first.  We can measure this in more detail at some future time.